File: FrontEnd/types/payment.ts
export interface CreatePaymentDto {
  OrderId: number;
  Amount: number;
  PaymentMethod: string;
}

export interface PaymentDto {
  Id: number;
  OrderId: number;
  Amount: number;
  PaymentMethod: string;
  TransactionId?: string | null;
  Status: string;
  PaidAt?: string | null;
  CreatedDate?: string | null;
}

File: ECommerce/Repository/Entities/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Repository.Entities
{
    public class Payment
    {
        public int Id { get; set; }

        public int OrderId { get; set; }
        public Order Order { get; set; }

        public string PaymentMethod { get; set; }
        public string TransactionId { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }
        public string Status { get; set; }
        public DateTime? PaidAt { get; set; }
        public DateTime? CreatedDate { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? UpdateDate { get; set; }
        public int? UpdateBy { get; set; }
    }
}